src/ray/gcs/gcs_client/gcs_client.cc

```diff
@@ -81,12 +81,29 @@ void GcsSubscriberClient::PubsubCommandBatch(
 GcsClient::GcsClient(const GcsClientOptions &options, UniqueID gcs_client_id)
     : options_(options), gcs_client_id_(gcs_client_id) {}
 
-Status GcsClient::Connect(instrumented_io_context &io_service) {
+Status GcsClient::Connect(instrumented_io_context &io_service,
+                          ClusterID const &cluster_id) {
+  cluster_token_promise_ = std::promise<ClusterID>();
```

Comment:
Add comments about why we have to use promise and do thing in async way instead of doing it in sync way.